packages/react-native-sdk/docusaurus/docs/reactnative/06-advanced/02-chat-with-video.mdx

```diff
@@ -3,6 +3,483 @@ title: Chat Integration
+
+To get the unread count of the Channel we need to focus primarily on watching the channel. So, to do it we will create a hook, that can be used easily:
+
+```tsx title="useChannelWatch.tsx"
+import { useCall } from '@stream-io/video-react-native-sdk';
+import { useEffect, useState } from 'react';
+import { useChatContext } from 'stream-chat-react-native';
+import type { Event } from 'stream-chat';
+
+/**
+ * This hook is responsible for creating the channel and watching it.
+ * It also makes sure to return the state whether the channel is being watched or not.
+ * @returns boolean
+ */
+export const useChannelWatch = () => {
+  const CHANNEL_TYPE = 'videocall';
+  const call = useCall();
+  const [channelWatched, setChannelWatched] = useState(false);
+  const { client } = useChatContext();
+  // The channel id is the same as the call id
+  const cid = `${CHANNEL_TYPE}:${call?.id}`;
+
+  useEffect(() => {
+    const channel = client.channel(CHANNEL_TYPE, call?.id);
+    channel.watch();
+
+    return () => {
+      channel.stopWatching();
+    };
+  }, [call?.id, client]);
+
+  useEffect(() => {
+    const handleEvent = (event: Event) => {
+      if (event?.cid === cid) {
+        setChannelWatched(true);
+      }
+    };
+
+    client.on('user.watching.start', handleEvent);
+
+    return () => {
+      client.off('user.watching.start', handleEvent);
+    };
+  }, [client, cid]);
+
+  return channelWatched;
+};
+```
+
+Here, we initially get the channel using the `client.channel` API and then, watch it using the `channel.watch()` function. To stop watching the channel, we use `channel.,stopWatching()` function.
+We listen to the `user.watching.start` event to set the channel watched state to `true` or `false`.
+
+Now we will create a hook to get the unread count value for a channel.
+
+```tsx title="useUnreadCount.tsx"
+import { useEffect, useState } from 'react';
+import { useChatContext } from 'stream-chat-react-native';
+import { Event } from 'stream-chat';
+import { useCall } from '@stream-io/video-react-native-sdk';
+
+type UseUnreadCountProps = {
+  channelWatched: boolean;
+};
+
+/**
+ * This hook is responsible for returning the unread count of the channel.
+ * This is done through listening to multiple events.
+ * @returns number
+ */
+export const useUnreadCount = ({ channelWatched }: UseUnreadCountProps) => {
+  const [unreadCount, setUnreadCount] = useState(0);
+  const CHANNEL_TYPE = 'videocall';
+  const { client } = useChatContext();
+  const call = useCall();
+
+  const cid = `${CHANNEL_TYPE}:${call?.id}`;
+
+  useEffect(() => {
+    if (!client) {
+      return;
+    }
+
+    const handleEvent = (event: Event) => {
+      if (event?.cid === cid) {
+        setUnreadCount(0);
+      }
+    };
+
+    client.on('notification.mark_read', handleEvent);
+    return () => client.off('notification.mark_read', handleEvent);
+  }, [client, cid]);
+
+  useEffect(() => {
+    if (!client || !channelWatched) {
```

Comment:
I dont see the use case of channelWatched prop actually..

if the channel is not watched, it wont be inside activeChannels and so handleevent will set a count of 0.

Reply:
It is just a security check to avoid bugs if the channel is not watched. Good to have it for now I think

Reply:
no please, lets make it simpler

Reply:
`just a security check to avoid bugs if the channel is not watched.`

we an add a note for that.. lets strive to make it simpler

Reply:
Removed the hook